docs/concepts/pull-dependencies.md

```diff
@@ -0,0 +1,202 @@
+# Pull Dependencies
+
+## Overview
+
+The interoperability of canisters on the Internet Computer (IC) is an important feature. 
+
+`dfx` provides a consistent developer workflow for integrating thrid party canisters.
+
+A service provider attaches [necessary metadata](canister-metadata.md) to the canister wasm.
+
+A service consumer then can pull dependencies directly from the IC mainnet and easily deploy them on a local replica.
+
+This document describes the consumer workflow and explains what happens behind the scene.
+
+## Workflow
+
+### 1. Declare "pull" dependencies in `dfx.json`
+
+Below is an example `dfx.json`, the service consumer is developing the "app" canister which has two pull dependencies:
+
+- "dep_b" which has canister ID of "yhgn4-myaaa-aaaaa-aabta-cai" on the mainnet.
+- "dep_c" which has canister ID of "yahli-baaaa-aaaaa-aabtq-cai" on the mainnet.
+
+```json
+{
+    "canisters": {
+        "app": {
+            "type": "motoko",
+            "main": "src/main.mo",
+            "dependencies": [
+                "dep_b", "dep_c"
+            ]
+        },
+        "dep_b": {
+            "type": "pull",
+            "id": "yhgn4-myaaa-aaaaa-aabta-cai"
+        },
+        "dep_c": {
+            "type": "pull",
+            "id": "yahli-baaaa-aaaaa-aabtq-cai"
+        }
+    }
+}
+```
+
+### 2. Pull the dependencies using `dfx deps pull`
+
+Running `dfx deps pull --network ic` will:
```

Comment:
Shouldn't `--network ic` be the default for `dfx deps pull`?  This would be the normal use case.  The other use cases would be `--network local` for CI, and `--network <some test net>` for test networks.

```suggestion
Running `dfx deps pull` will:
```